tensorflow/contrib/rnn/python/ops/rnn_cell.py

```diff
@@ -2352,52 +2332,48 @@ def call(self, inputs, state):
       for group_id in range(self._number_of_groups):
         with vs.variable_scope("group%d" % group_id):
           x_g_id = array_ops.concat(
-              [
-                  self._get_input_for_group(inputs, group_id,
-                                            self._group_shape[0]),
-                  self._get_input_for_group(m_prev, group_id,
-                                            self._group_shape[0])
-              ],
-              axis=1)
-          if self._linear1 is None:
-            self._linear1 = _Linear(x_g_id, 4 * self._group_shape[1], False)
-          R_k = self._linear1(x_g_id)  # pylint: disable=invalid-name
+            [self._get_input_for_group(inputs, group_id,
+                                       int(input_size / self._number_of_groups)),
+             self._get_input_for_group(m_prev, group_id,
+                                       int(self._output_size / self._number_of_groups))], axis=1)
+          if self._linear1[group_id] is None:
+            self._linear1[group_id] = _Linear(x_g_id, 4 * self._group_shape[1], False)
```

Comment:
group_shape[0] is never used. And after the change, not clear why is it needed at all